### DISCUSSION TITLE
The same library among different OS

### DISCUSSION THREAD

**Discussion by LukashonakV** [mesonbuild/meson#11514] [Q&A] [answered]
The issue here: iniparser library. 
It's works well
```meson
iniparser_dep = compiler.find_library('iniparser', has_headers: 'iniparser.h', required: false)
```
But in Gentoo exact this library header for particular reasons is stored in /usr/include/iniparser4
So in Gentoo
```meson
iniparser_dep = compiler.find_library('iniparser4', has_headers: 'iniparser4/iniparser.h', required: false)
```
And issue in Gentoo is ... due to source code does have : #include <iniparser.h> it doesn't compile... 
So it could be solved 
1. Using ifdef directive in the code... or
2.  ```meson include_directories: [include_directories('input'), include_directories('output') , '/usr/include/iniparser4'],```

I'm trying to do the second point and just redefine a little inlcude_directories... But I have stuck... 
Is there a way to extract full path from find_library ? Or what is a proper way to get include directory which is using by find_library in order to fix it a little and provide it further ?

**Comment by eli-schwartz** [comment]
Your source code could `#include <iniparser4/iniparser.h>` and use a has_header check to see which include style to use.

In fact, you can also use `__has_include(<iniparser4/iniparser.h>)` to do this check without running additional meson.build checks. It's a vendor extension for GCC/clang, and standardized as part of C++17.

...

Upstream iniparser has rejected numerous offers to improve the Makefile just to actually install the built library and headers, which is why the installation directory is totally unspecified. There's a pkg-config file but that isn't installed either and hardcodes /usr/lib. No one seems to actually install it anyway.

**Comment by LukashonakV** [comment, marked as answer]
Hi @eli-schwartz , thank you
Btw
```meson
########################
# Checking for iniparser
########################
iniparser_dep = compiler.find_library('iniparser', has_headers: 'iniparser.h', required: false)

if not iniparser_dep.found()
  iniparser4_dep = compiler.find_library('iniparser4', has_headers: 'iniparser4/iniparser.h', required: false)

   iniparser_dep = declare_dependency(
      include_directories: include_directories('/usr/include/iniparser4'),
      dependencies: iniparser4_dep)
endif

if not iniparser_dep.found()
   error('iniparser library is required!')
else
   if not compiler.has_function('iniparser_load', dependencies: iniparser_dep)
      error('iniparser library is required!')
   endif
endif
```

Actually meson is able to provide right header.  Do you know is it possible in such case to avoid of hardcodes
include_directories: include_directories('/usr/include/iniparser4'),
and get the it somehow from dependency...

**Reply by eli-schwartz** [reply]
The problem is that meson doesn't know it's in /usr/include at all. All it knows is that the compiler was able to find that header, possibly in its internal include directories.

That's why I suggested actually using the same include style in your source code that you want to use for the meson.build checks.

**Reply by LukashonakV** [reply]
Thank you a lot @eli-schwartz  . Now I got it
